How-to-upgrade-review-version.md

```diff
@@ -0,0 +1,399 @@
+    "noEmit": true,
+    "resolveJsonModule": true
+    "esModuleInterop": true
+    "strict": false,
+  },
+  "exclude": [
+    "tests/fixtures",
+    "tests/integration"
+  ]
+}
+```
+
+For Javascript, your "compilerOptions" will pretty much always look
+like this. "resolveJsonModule" is optional, but you'll need it if your
+code ever `require`s a JSON file, so that Typescript will analyze it
+too. *Officially*, strict should be false, but as you will see later,
+ strict mode can be pretty great for Javascript code, not just
+ Typescript code.
+
+For typescript-eslint-parser, I added an exclude list. I wanted to
+check *all* the source files, because it's nice to have high
+aspirations. But it turns out the parser's tests take the form of JS
+files themselves, and I don't want to check those. Most of them are
+malformed in some way.
+
+You might want to use "include" if, say, you only want to check your
+source and not your tests or scripts. Or you can use "files" to give
+an explicit list of files to use, but this is annoying except for
+small projects.
+
+OK, you're all set. Run tsc and make sure it prints out errors. Now open up
+files in your editor and make sure the same errors show up there.
+
+Congratulations! You've done the only *required* part. Everything else
+just helps reduce the number of errors you see when editing.
+
+## Install @types packages.
+
+Your first order of business is to install types for packages you use.
+This is the easiest way to reduce the number of errors. Basically, if
+you use package X, and you see errors when you use it, try installing
+`@types/X`. Tons of packages have definitions, so you will probably
+find most of your dependencies are typed.
+
+Note that if you use node at all, you likely need `@types/node`.
+
+For typescript-eslint-parser, I installed:
+
+* @types/node
+* @types/jest
+* @types/estree
+* @types/shelljs
+* @types/eslint-scope
+
+If you are really, truly, completely in stealth mode, make sure not to
+save these installations in package.json. That's a bit extreme,
+though, isn't it?
+
+Anyway, these three types packages still didn't work:
+
+* @types/shelljs
+* @types/estree
+* @types/eslint-scope
+
+They fail for different reasons, though. shelljs and es-lint-scope are
+just missing whole chunks of the package's types. estree *has* all the
+correct types, but the types aren't imported correctly. So now you can
+decide where to start fixing errors.
+
+You can
+
+1. Add missing types in dependencies.
+2. Fix references to types in dependencies.
+4. Add missing types in your own code.
+3. Work around missing types in dependencies.
+5. Fix errors in existing types.
+6. Add type annotations to everything else.
+
+## Add missing types in dependencies
+
+Let's start with @types/shelljs. In Makefile.js, I see a few errors.
+The first is that the module `require('shelljs/make')` isn't found.
+The second group of errors is that the names 'find', 'echo' and a few
+others aren't found.
```

Comment:
I was confused here because when I ran tsc locally I didn't get these errors. It turns that's because you already checked in the definitions to DefinitelyTyped. I think you should mention this in case people wonder why they're not seeing the errors. Also, you might want to print out the errors (as of the current compiler version) so people know what they're actually trying to fix.

Reply:
Good idea.